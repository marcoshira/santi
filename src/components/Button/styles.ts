import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    background: inherit;
    color: ${theme.colors.teal};
    /* border: 0.5px solid ${theme.colors.teal}; */
    border: none;
    font-size: ${theme.font.sizes.ltmedium};
    font-weight: 500;
    margin-left: ${theme.spacings.xxsmall};
    margin-top: 5px;
    margin-bottom: 4px;
    position: relative;
    transition: all 300ms ease-in-out;

    &:disabled {
      background: ${theme.colors.mediumGrey};
      color: ${theme.colors.white};
      border: 2px solid ${theme.colors.mediumGrey};
      transition: none;

      &:hover {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.mediumGrey};
        cursor: not-allowed;
      }

      &::after {
        content: none;
      }
    }
    &:hover {
      color: ${theme.colors.thirdColor};
      cursor: pointer;
    }

    &::after {
      content: '';
      position: absolute;
      bottom: 0.2rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.thirdColor};
      transition: all 200ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
