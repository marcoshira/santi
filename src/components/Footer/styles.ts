import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGrey};

    a {
      color: inherit;
      text-decoration: none;
    }
  `}
`;
