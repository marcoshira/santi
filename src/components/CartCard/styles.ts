import styled, { css } from 'styled-components';
import { Wrapper as Btn } from '../Button/styles';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;
    overflow-x: hidden;
    box-shadow: 3px 3px 15px ${theme.colors.teal};
    border-radius: 15px;
    color: ${theme.colors.navy};
    background: ${theme.colors.footerWhite};
    position: relative;
    padding: ${theme.spacings.small} 0;
    margin-bottom: ${theme.spacings.ltmedium};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      min-height: 50rem;
      width: 85%;
      justify-content: center;
      align-items: center;
      margin: ${theme.spacings.medium} auto;
    }

    a {
      cursor: pointer;
      &:hover {
        img {
          opacity: 0.6;
        }
      }
    }

    img {
      border-radius: 10px;
      height: 20rem;
      transition: ${theme.transitions.faster};

      @media ${theme.media.lteMedium} {
        height: 15rem;
      }
    }

    ${Title} {
      margin: 0;
    }
  `}
`;

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    width: 55%;
    a {
      text-decoration: none;
      color: ${theme.colors.navy};
      transition: ${theme.transitions.faster};
      &:hover {
        color: ${theme.colors.thirdColor};
      }
    }

    @media ${theme.media.lteMedium} {
      width: 100%;
      padding: ${theme.spacings.medium};
    }
  `}
`;

export const ImageWrapper = styled.div`
  display: flex;
  justify-content: center;
  width: 45%;
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    margin-top: 2rem;
    margin-right: 2rem;

    ${Btn} {
      min-width: 48%;

      margin: 0rem;
      margin-right: 1rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.footerWhite};
      color: ${theme.colors.teal};

      &:hover {
        border: 2px solid ${theme.colors.thirdColor};
        background: ${theme.colors.footerWhite};
        color: ${theme.colors.thirdColor};
      }

      &::after {
        content: none;
      }

      :nth-of-type(2) {
        background: ${theme.colors.teal};
        color: ${theme.colors.footerWhite};

        &:hover {
          border: 2px solid ${theme.colors.teal};
          background: ${theme.colors.hoverTeal};
        }
      }
    }

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      margin-right: 0rem;

      ${Btn} {
        width: 100%;
        margin: ${theme.spacings.tiny} 0;
      }
    }
  `}
`;
