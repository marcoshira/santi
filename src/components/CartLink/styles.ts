import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: absolute;
    left: 40rem;
    top: 9rem;
    width: calc(100vw - 55rem);

    @media ${theme.media.lteMedium} {
      width: 100%;
      left: auto;
      top: 6rem;
      margin: 0 auto;

      > ${Title} {
        margin-left: ${theme.spacings.ltlarge};
      }
    }
  `}
`;
