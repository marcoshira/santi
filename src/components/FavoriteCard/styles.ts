import styled, { css } from 'styled-components';
import { Wrapper as Btn } from '../Button/styles';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;
    box-shadow: 3px 3px 15px ${theme.colors.teal};
    border-radius: 15px;
    color: ${theme.colors.navy};
    background: ${theme.colors.footerWhite};
    position: relative;
    padding: ${theme.spacings.small} 0;
    margin-bottom: ${theme.spacings.ltmedium};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      min-height: 53rem;
      width: 85%;
      justify-content: center;
      align-items: center;
      margin: ${theme.spacings.medium} auto;
      padding-bottom: 0;
    }

    a {
      cursor: pointer;
      &:hover {
        img {
          opacity: 0.6;
        }
      }
    }

    img {
      border-radius: 10px;
      height: 20rem;
      transition: ${theme.transitions.faster};

      @media ${theme.media.lteMedium} {
        height: 15rem;
      }
    }

    ${Title} {
      margin: 0;
    }
  `}
`;

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    width: 55%;
    a {
      text-decoration: none;
      color: ${theme.colors.navy};
      transition: ${theme.transitions.faster};
      &:hover {
        color: ${theme.colors.thirdColor};
      }
    }
    svg {
      height: 3rem;
      position: absolute;
      top: 2rem;
      right: 2rem;
      color: ${theme.colors.teal};
      cursor: pointer;
      transition: ${theme.transitions.fast};

      &:hover {
        color: ${theme.colors.hoverTeal};
      }
    }

    @media ${theme.media.lteMedium} {
      width: 100%;
      padding: ${theme.spacings.medium};
    }
  `}
`;

export const ImageWrapper = styled.div`
  ${({ theme }) => css`
    width: 45%;
    display: flex;
    justify-content: center;

    img {
    }
  `}
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-around;
    flex-direction: row;
    margin-top: 3rem;
    ${Btn} {
      min-width: 47%;

      margin: 0rem;
      margin-right: 1rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.footerWhite};
      color: ${theme.colors.teal};

      &:hover {
        border: 2px solid ${theme.colors.thirdColor};
        background: ${theme.colors.footerWhite};
        color: ${theme.colors.thirdColor};
      }

      &::after {
        content: none;
      }

      :nth-of-type(2) {
        background: ${theme.colors.teal};
        color: ${theme.colors.footerWhite};

        &:hover {
          border: 2px solid ${theme.colors.teal};
          background: ${theme.colors.hoverTeal};
        }
      }
    }

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      margin-right: 0rem;

      ${Btn} {
        width: 100%;
        margin: ${theme.spacings.tiny} 0;
      }
    }
  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 50%;
    justify-content: left;
    flex-flow: row nowrap;
    font-size: 1.6rem;
    margin-top: 3rem;

    @media ${theme.media.lteMedium} {
      width: 100%;
      margin-top: 1rem;
    }

    label {
      margin: 8px ${theme.spacings.ltlarge} 0 0;
      color: ${theme.colors.darkGrey};

      @media ${theme.media.lteMedium} {
        margin: 7px ${theme.spacings.hero} 0 0;
      }
    }

    input {
      width: 40px;
      display: flex;
      text-align: center;
      align-items: center;
      justify-content: center;
      border: 1px solid ${theme.colors.grey};
      height: 35px;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    padding: 0 10px;
    background: ${theme.colors.footerWhite};
    border: 1px solid ${theme.colors.grey};
  `}
`;
