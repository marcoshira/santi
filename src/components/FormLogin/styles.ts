import styled, { css } from 'styled-components';
import { Wrapper as Button } from '../Button/styles';

export const Wrapper = styled.form``;

export const ButtonWrapper = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.medium};
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    ${Button} {
      a {
        text-decoration: none;
        color: ${theme.colors.footerWhite};
      }
      width: 15rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.teal};
      color: ${theme.colors.footerWhite};
      border-radius: ${theme.spacings.tiny};
      margin-left: 0;
      &:hover {
        background: ${theme.colors.hoverTeal};
      }
      &::after {
        height: 0;
      }
      &:disabled {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.footerWhite};
        border: 2px solid ${theme.colors.mediumGrey};

        &:hover {
          background: ${theme.colors.mediumGrey};
          color: ${theme.colors.white};
          border: 2px solid ${theme.colors.mediumGrey};
          cursor: not-allowed;
        }

        &::after {
          content: none;
        }
      }

      @media ${theme.media.lteSmall} {
        width: 100%;
        padding: ${theme.spacings.xsmall} ${theme.spacings.xsmall};
      }
    }
  `}
`;
