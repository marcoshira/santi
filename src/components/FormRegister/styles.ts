import styled, { css } from 'styled-components';
import { Wrapper as Button } from '../Button/styles';

export const Wrapper = styled.form`
  ${({ theme }) => css``}
`;

export const ButtonWrapper = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.medium};
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    ${Button} {
      a {
        text-decoration: none;
        color: ${theme.colors.footerWhite};
      }
      width: 15rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.teal};
      color: ${theme.colors.footerWhite};
      border-radius: ${theme.spacings.tiny};
      margin-left: 0;
      &:hover {
        background: ${theme.colors.hoverTeal};
      }
      &::after {
        height: 0;
      }
      &:disabled {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.footerWhite};
        border: 2px solid ${theme.colors.mediumGrey};

        &:hover {
          background: ${theme.colors.mediumGrey};
          color: ${theme.colors.white};
          border: 2px solid ${theme.colors.mediumGrey};
          cursor: not-allowed;
        }

        &::after {
          content: none;
        }
      }
    }
  `}
`;

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.thirdColor};
    font-size: ${theme.font.sizes.normal};
    margin-top: ${theme.spacings.small};
  `}
`;
