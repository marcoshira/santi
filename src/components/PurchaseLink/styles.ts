import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: absolute;
    left: 40rem;
    top: 9rem;
    width: calc(100vw - 50rem);

    @media ${theme.media.lteMedium} {
      width: 100%;
      left: auto;
      top: 6rem;
      margin: 0 auto;

      > ${Title} {
        margin-left: ${theme.spacings.ltlarge};
      }
      p {
        margin-left: ${theme.spacings.ltlarge};
      }
    }
  `}
`;
