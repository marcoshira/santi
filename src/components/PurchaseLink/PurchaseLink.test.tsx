import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PurchaseLink } from '.';
import { PurchaseAttributesProps } from '../../shared-types/shared-types';
import mock from '../ProfileComponent/mock';

const teste = {
  Item: [],
};

const props: PurchaseAttributesProps =
  mock.usersPermissionsUser.data.attributes.puchase.data.attributes;

describe('<PurchaseLink />', () => {
  it('should render', () => {
    const { container } = renderTheme(<PurchaseLink {...props} />);

    expect(
      screen.getByRole('heading', { name: 'Purchases' }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('should render without purchases', () => {
    renderTheme(<PurchaseLink {...teste} />);

    expect(screen.getByText('No Purchases Yet')).toBeInTheDocument();
  });
});
