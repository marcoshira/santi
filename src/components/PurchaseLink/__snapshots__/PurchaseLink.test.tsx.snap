// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PurchaseLink /> should render 1`] = `
.c2 {
  color: inherit;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c8 {
  padding: 0.8rem 1rem;
  background: inherit;
  color: #8c4bff;
  border: none;
  font-size: 2rem;
  font-weight: 500;
  margin-left: 0.8rem;
  margin-top: 5px;
  margin-bottom: 4px;
  position: relative;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:disabled {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
  -webkit-transition: none;
  transition: none;
}

.c8:disabled:hover {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
  cursor: not-allowed;
}

.c8:disabled::after {
  content: none;
}

.c8:hover {
  color: #ff4a4a;
  cursor: pointer;
}

.c8::after {
  content: '';
  position: absolute;
  bottom: 0.2rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #ff4a4a;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c8:hover::after {
  left: 25%;
  width: 50%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  box-shadow: 3px 3px 15px #8c4bff;
  border-radius: 15px;
  color: #0A1128;
  background: #FFFFFF;
  position: relative;
  padding: 1.6rem 0;
  margin-bottom: 2rem;
}

.c3 #close {
  position: absolute;
  top: 2rem;
  right: 2rem;
}

.c3 #close svg {
  height: 3rem;
}

.c3 a {
  cursor: pointer;
}

.c3 a:hover img {
  opacity: 0.6;
}

.c3 img {
  border-radius: 10px;
  height: 20rem;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c3 .c1 {
  margin: 0;
}

.c3 .c7 {
  position: absolute;
  bottom: 2rem;
  right: 4rem;
  padding: 1rem 2.8rem;
  border: 2px solid #8c4bff;
  background: #8c4bff;
  color: #FFFFFF;
  border-radius: 0.4rem;
  margin-left: 0;
}

.c3 .c7 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
}

.c3 .c7:hover {
  background: #6d39c7;
}

.c3 .c7::after {
  height: 0;
}

.c3 .c7:disabled {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
}

.c3 .c7:disabled:hover {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
  cursor: not-allowed;
}

.c3 .c7:disabled::after {
  content: none;
}

.c5 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0A1128;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c5 a:hover {
  color: #ff4a4a;
}

.c5 #date {
  font-size: 1.6rem;
  color: rgb(141,141,144);
  position: absolute;
  bottom: 0.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 45%;
}

.c0 {
  position: absolute;
  left: 40rem;
  top: 9rem;
  width: calc(100vw - 50rem);
}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    min-height: 45rem;
    width: 85%;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding-bottom: 8.0rem;
    margin: 2.4rem auto;
  }
}

@media (max-width:768px) {
  .c3 img {
    height: 15rem;
  }
}

@media (max-width:768px) {
  .c3 .c7 {
    left: 2rem;
    right: 2rem;
  }
}

@media (max-width:768px) {
  .c5 {
    width: 100%;
    padding: 2.4rem;
  }
}

@media (max-width:768px) {
  .c5 #date {
    bottom: 6.5rem;
  }
}

@media (max-width:768px) {
  .c4 {
    width: auto;
  }
}

@media (max-width:768px) {
  .c0 {
    width: 100%;
    left: auto;
    top: 6rem;
    margin: 0 auto;
  }

  .c0 > .c1 {
    margin-left: 2.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      Purchases
    </h1>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <a
          href="/products/2"
        >
          <img
            src="https://res.cloudinary.com/djqrityvi/image/upload/v1652740674/ps5_8a252e1eaa.jpg"
          />
        </a>
      </div>
      <div
        class="c5"
      >
        <a
          href="/products/2"
        >
          <h1
            class="c1 c2"
          >
            PlayStation 5
          </h1>
        </a>
        <a
          aria-label="close"
          id="close"
        >
          <svg
            aria-hidden="true"
            class="c6"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
            />
          </svg>
        </a>
        <p>
          Price: R$ 
          4799.99
           | Quantity: 
          2
        </p>
        <p>
          Total price: R$: 
          9599.98
        </p>
        <p
          id="date"
        >
          Bought at 
          2022-04-30
        </p>
        <a
          href="/products/2"
        >
          <button
            class="c7 c8"
          >
            Buy Again
          </button>
        </a>
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <a
          href="/products/2"
        >
          <img
            src="https://res.cloudinary.com/djqrityvi/image/upload/v1652740674/ps5_8a252e1eaa.jpg"
          />
        </a>
      </div>
      <div
        class="c5"
      >
        <a
          href="/products/2"
        >
          <h1
            class="c1 c2"
          >
            PlayStation 5
          </h1>
        </a>
        <a
          aria-label="close"
          id="close"
        >
          <svg
            aria-hidden="true"
            class="c6"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
            />
          </svg>
        </a>
        <p>
          Price: R$ 
          4799.99
           | Quantity: 
          2
        </p>
        <p>
          Total price: R$: 
          9599.98
        </p>
        <p
          id="date"
        >
          Bought at 
          2022-04-30
        </p>
        <a
          href="/products/2"
        >
          <button
            class="c7 c8"
          >
            Buy Again
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
